### PR TITLE
Improve cen_route_entries use common method.

### PR DESCRIPTION
**Pull request by Lexsss**

### CODE REVIEW

File: alicloud/data_source_alicloud_cen_route_entries_test.go [hunk 1]
```diff
@@ -1,69 +1,41 @@
...
+			"instance_id":    `"${alicloud_cen_route_entry.default.instance_id}"`,
+			"route_table_id": `"${alicloud_cen_route_entry.default.route_table_id}"`,
+			"cidr_block":     `"11.0.0.0/16"`,
+		}),
+		fakeConfig: testAccAlicloudCenRouteEntriesDataSourceConfig(rand, map[string]string{
+			"instance_id":    `"${alicloud_cen_route_entry.default.instance_id}_fake"`,
+			"route_table_id": `"${alicloud_cen_route_entry.default.route_table_id}"`,
+			"cidr_block":     `"11.1.0.0/16"`,
+		}),
+	}
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_cen_route_entries_test.go]: please complete test case of when no cidr_block

File: alicloud/import_alicloud_cen_route_entry_test.go [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckCenRouteEntryDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCenRouteEntryConfig(EcsInstanceCommonTestCase, defaultRegionToTest),
+				Config: testAccCenRouteEntryConfig(rand),
 			},
 
 			{
```
Comment (YY-tomorrow) [inline alicloud/import_alicloud_cen_route_entry_test.go]: please ues resource steps

File: alicloud/resource_alicloud_cen_route_entry.go [hunk 3]
```diff
@@ -60,27 +58,27 @@ func resourceAlicloudCenRouteEntryCreate(d *schema.ResourceData, meta interface{
...
 			}
 			return resource.NonRetryableError(err)
 		}
-
+		addDebug(request.GetActionName(), raw)
 		return nil
 	})
 	if err != nil {
-		return fmt.Errorf("Publish CEN route entry timeout, CEN %s child instance %s vtb %s cidr %s, error info: %#v.", cenId, childInstanceId, vtbId, cidr, err)
+		return WrapErrorf(err, DataDefaultErrorMsg, "alicloud_cen_route_entry", request.GetActionName(), AlibabaCloudSdkGoERROR)
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_cen_route_entry.go]: please use DefaultErrorMsg

File: alicloud/resource_alicloud_cen_route_entry.go [hunk 4]
```diff
@@ -90,30 +88,29 @@ func resourceAlicloudCenRouteEntryRead(d *schema.ResourceData, meta interface{})
...
-	if len(parts) != 3 {
-		return fmt.Errorf("invalid resource id")
+	parts, err := ParseResourceId(d.Id(), 3)
+	if err != nil {
+		return WrapError(err)
 	}
 	cenId := parts[0]
 
-	resp, err := cenService.DescribePublishedRouteEntriesById(d.Id())
+	response, err := cenService.DescribeCenRouteEntry(d.Id())
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_cen_route_entry.go]: response change to object

File: alicloud/resource_alicloud_cen_route_entry.go [hunk 5]
```diff
@@ -142,28 +139,24 @@ func resourceAlicloudCenRouteEntryDelete(d *schema.ResourceData, meta interface{
 	request.DestinationCidrBlock = cidr
 
 	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
-		_, err := client.WithCenClient(func(cbnClient *cbn.Client) (interface{}, error) {
+		raw, err := client.WithCenClient(func(cbnClient *cbn.Client) (interface{}, error) {
 			return cbnClient.WithdrawPublishedRouteEntries(request)
 		})
 		if err != nil {
 			if IsExceptedErrors(err, []string{NotFoundRoute, InstanceNotExistMsg}) {
 				return nil
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_cen_route_entry.go]: please remove 146-147 to 157

File: alicloud/service_alicloud_cen.go [hunk 6]
```diff
@@ -404,40 +404,44 @@ func (s *CenService) DescribePublishedRouteEntriesById(id string) (c cbn.Publish
...
-			return err
+			return WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
 		}
-		resp, _ := raw.(*cbn.DescribePublishedRouteEntriesResponse)
-		if resp == nil || len(resp.PublishedRouteEntries.PublishedRouteEntry) <= 0 {
-			return GetNotFoundErrorFromString(GetNotFoundMessage("CEN RouteEntries", id))
+		addDebug(request.GetActionName(), raw)
+		response, _ := raw.(*cbn.DescribePublishedRouteEntriesResponse)
+		if len(response.PublishedRouteEntries.PublishedRouteEntry) <= 0 {
+			return WrapErrorf(Error(GetNotFoundMessage("CenRouteEntries", cenId)), NotFoundMsg, ProviderERROR)
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_cen.go]: why use cenId

File: alicloud/service_alicloud_cen.go [hunk 7]
```diff
@@ -404,40 +404,44 @@ func (s *CenService) DescribePublishedRouteEntriesById(id string) (c cbn.Publish
...
+func (s *CenService) WaitForCenRouterEntry(id string, status Status, timeout int) error {
+	deadline := time.Now().Add(time.Duration(timeout) * time.Second)
 
 	for {
-		routeEntry, err := s.DescribePublishedRouteEntriesById(id)
+		object, err := s.DescribeCenRouteEntry(id)
 		if err != nil {
-			return nil
+			if NotFoundError(err) {
+				if status == NOPUBLISHED {
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_cen.go]: why use NOPUBLISHED

File: alicloud/resource_alicloud_cen_route_entry_test.go [hunk 8]
```diff
@@ -109,31 +72,58 @@ func testAccCenRouteEntryConfig(common, region string) string {
...
+
+		routeEntryItem, err := cenService.DescribeCenRouteEntry(rs.Primary.ID)
+		if err != nil {
+			if NotFoundError(err) {
+				continue
+			}
+			return err
+		}
+
+		if routeEntryItem.PublishStatus == string(NOPUBLISHED) {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cen_route_entry_test.go]: why？

**Pull request merged**